lua/nvim-tree/renderer/components/padding.lua

```diff
@@ -25,30 +25,30 @@ local function get_padding_indent_markers(depth, idx, nodes_number, markers, wit
 
   if depth > 0 then
     local has_folder_sibling = check_siblings_for_folder(node, with_arrows)
-    local rdepth = depth / 2
-    markers[rdepth] = idx ~= nodes_number
-    for i = 1, rdepth do
+    local indent = string.rep(" ", M.config.indent_width - 1)
+    markers[depth] = idx ~= nodes_number
+    for i = 1, depth do
       local glyph
-      if idx == nodes_number and i == rdepth then
+      if idx == nodes_number and i == depth then
         glyph = M.config.indent_markers.icons.corner
-      elseif markers[i] and i == rdepth then
+      elseif markers[i] and i == depth then
         glyph = M.config.indent_markers.icons.item
       elseif markers[i] then
         glyph = M.config.indent_markers.icons.edge
       else
         glyph = M.config.indent_markers.icons.none
       end
 
-      if not with_arrows or (inline_arrows and (rdepth ~= i or not node.nodes)) then
-        padding = padding .. glyph .. " "
+      if not with_arrows or (inline_arrows and (depth ~= i or not node.nodes)) then
+        padding = padding .. glyph .. indent
       elseif inline_arrows then
         padding = padding
-      elseif idx == nodes_number and i == rdepth and has_folder_sibling then
-        padding = padding .. base_padding .. glyph .. "── "
-      elseif rdepth == i and not node.nodes and has_folder_sibling then
-        padding = padding .. base_padding .. glyph .. " " .. base_padding
+      elseif idx == nodes_number and i == depth and has_folder_sibling then
+        padding = padding .. base_padding .. glyph .. "──" .. indent
```

Comment:
We should extend this and the corner:

![20220814_155144 497x927](https://user-images.githubusercontent.com/1505378/184524403-9f3e9012-feea-4065-9893-7932f4743dc3.png)

A `renderer.indent_markers.icons.bottom = "─"` should be added.

Reply:
Done, I've added `icons.bottom` char and added support for corner extension in all three possible option combinations (inline/non-inline arrows, no arrows). Without icons it looks like this:

| `inline_arrows = true` | `inline_arrows = false` | `folder_arrow = false` |
|---|---|---|
|![image](https://user-images.githubusercontent.com/5783792/184607110-092fe1bd-2c42-4cdd-a97c-8b92ecdf1e40.png)|![image](https://user-images.githubusercontent.com/5783792/184607564-ebf1b9fa-e297-483b-a8dd-d276121de9d4.png)|![image](https://user-images.githubusercontent.com/5783792/184608062-53f3bf54-3ee1-4e58-80e4-c6e80394a08a.png)|